Backend/app/therapy_sessions/serializers.py

```
@@ -0,0 +1,738 @@
+        if value is None:
+            return value
+            
+        try:
+            patient = User.objects.get(id=value, user_type='patient')
+            # Check if patient is connected to the therapist
+            therapist = self.context['request'].user
+            if not hasattr(patient, 'patient_profile') or not patient.patient_profile.therapist or patient.patient_profile.therapist.user != therapist:
+                raise serializers.ValidationError("Patient is not connected to this therapist.")
+            return value
+        except User.DoesNotExist:
+            raise serializers.ValidationError("Patient not found.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)